backend/waste-optimizer/routes.py

```
@@ -0,0 +1,378 @@
+    raw_report: str = Field(..., description="Unstructured text report containing information on food surplus and demand.")
+    priority_focus: str = Field(default="all", description="Focus area: 'hunger_relief', 'farmer_support', 'environment', or 'all'")
+
+class AgentOutput(BaseModel):
+    allocation_plan: str
+    human_summary: str
+    estimated_impact: Dict[str, Any]
+
+# --- Set up the Gemini API ---
+GEMINI_API_KEY = os.getenv("GEMINI_API_KEY")
+if not GEMINI_API_KEY:
+    raise ValueError("GEMINI_API_KEY environment variable not found. Please set it in your .env file.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)